src/qubex/simulator/deprecated/quantum_simulator.py

```
@@ -0,0 +1,592 @@
+        substates = [state.ptrace(index) for state in self.states]
+
+        if frame == "qubit":
+            # rotate the states to the qubit frame
+            times = self.control.times
+            qubit = self.system.get_object(label)
+            f_drive = self.control.frequency
+            f_qubit = qubit.frequency
+            delta = 2 * np.pi * (f_qubit - f_drive)
+            dim = qubit.dimension
+            N = qt.num(dim)
+            U = lambda t: (-1j * delta * N * t).expm()
```

Ruff (0.8.2)

Do not assign a `lambda` expression, use a `def`

Rewrite `U` as a `def`

(E731)